cognite/extractorutils/base.py

```
@@ -210,37 +212,53 @@ def recursive_find_state_store(d: dict[str, Any]) -> StateStoreConfig | None:
 
-    def _report_success(self) -> None:
-        """
-        Called on a successful exit of the extractor
-        """
+    def _report_run(self, status: str, message: str) -> None:
+        MAX_MESSAGE_LENGTH_FOR_EXTRACTION_PIPELINE_RUN = 1000
         if self.extraction_pipeline:
-            self.logger.info("Reporting new successful run")
-            self.cognite_client.extraction_pipelines.runs.create(
-                ExtractionPipelineRun(
+            try:
+                message = message or ""
+                shortened_message = shorten(
+                    message,
+                    width=MAX_MESSAGE_LENGTH_FOR_EXTRACTION_PIPELINE_RUN,
+                    placeholder="...",
+                )
+
+                run = ExtractionPipelineRun(
                     extpipe_external_id=self.extraction_pipeline.external_id,
-                    status="success",
-                    message=self.success_message,
+                    status=status,
+                    message=shortened_message,
                 )
-            )
 
-    def _report_error(self, exception: BaseException) -> None:
+                self.logger.info(f"Reporting new {status} run: {message}")
+                self.cognite_client.extraction_pipelines.runs.create(run)
+            except CogniteAPIError as e:
+                self.logger.exception(
+                    f"error while reporting run - status {status} - message {message} . Error: {str(e)}"
+                )
+
+    def _report_success(self, message: str | None = None) -> None:
+        """
+        Called on a successful exit of the extractor
+        """
+        message = message or self.success_message
+        self._report_run("success", message)
+
+    def _report_error(self, exception: BaseException, message: str | None = None) -> None:
         """
         Called on an unsuccessful exit of the extractor
 
         Args:
             exception: Exception object that caused the extractor to fail
```

You are missing doc comments for `message`. Also, you should probably make it clear that if `message` is missing, then `exception` will be used to create the error.

Also, perhaps, make it clear that if you provide a message, the exception is ignored. This isn't the greatest API, IMO.

---

Naah you are right. 
Having two sources for message makes it confusing. I would rather split it up, if someone wants to report an error and they pass an exception, we construct the message. 
And if someone wants to directly provide the message and report a failure we should just have a _report_failure method that essentially creates a failure run.